### ISSUE TITLE
Decameron Project Tasks!

### ISSUE THREAD

**Issue by ebeshero** [jlm323/DecameronProject#7]
- [x] Analyze the Oakland Team's markup in the engDecameron file. **ARE** they capturing the nested narratives? If **not**, do we want to keep their structural markup, or change it?
- [x] Look for the original source of their code on the Brown Decameron Project. 
- [x] Articulate the research question that will control your markup strategy! 
- [x] By Friday, 2/26: Everybody, READ a couple of sets of nested narratives, in oXygen. Survey Day 1, and look for stories within stories, to see how they're signalled. Tell us about where to find these when you spot them! (If you can, look at another Day (like Day 10) to survey. Each person should try to closely read stories over a few days of your choice!
- It's very easy for us to rip out unwanted tags, so don't be anxious about the markup that's already here if it doesn't make sense.

**Comment by RJP43** [comment]
**Research Ideas proposed in class 2/24:**  
_10 days - 10 stories a day_  
What marks a separation between days of the entire narrative?  
What marks a separation between stories of a single day?  
What levels of structure matter: stories, paragraphs, sentences?  
Where is there overlap?  
How are characters associated with places within a structural unit?  
What stories do characters appear in?  
What stories do places appear in?  
What stories do these characters and places overlap?  
What stories do characters overlap?  
What stories do places overlap?

**Comment by ebeshero** [comment]
I am going to take a look at the Project Gutenberg file and see if it's formatted differently in a way that helps to signal the nested narratives better than this Obdurodon/Brown file. @ebeshero

**Comment by ebeshero** [comment]
Decameron team! What is happening with the project? @jlm323 @mmm202 @laurenmcguigan @mmm202

**Comment by mmm202** [comment]
@jlm323 @mjb232 @laurenmcguigan Have we picked what each of us can do?

**Comment by jlm323** [comment]
We haven't decided who is doing what yet but I think a good place to start is doing the [project midterm](https://github.com/ebeshero/DHClass-Hub/issues/161). Once we've started looking at the Decameron file individually and marking it up, I say we get together and discuss what we've found and start assigning tasks. I think we can draw from the markup each of us does and go from there.

**Comment by laurenmcguigan** [comment]
Sorry I haven't been able to do much for on this yet, I've been working all week. But I am off tomorrow, so I can get started on the schematron if you guys want me to. Should we all be doing individual ones or should we make one together? perhaps we can each come up with at least one rule and post it here? And I can also try to extract names and places. Can I do that using find and replace? Or should I use XSLT to extract them and turn them into html?

**Comment by mmm202** [comment]
I think we need to at least start our markup before Monday and write the schematron. What ever way everyone wants to do it though is fine with me.

**Comment by RJP43** [comment]
Hey everyone @mmm202 @mjb232 @jlm323 @laurenmcguigan ,   

I want to offer some advice on a workflow for the project, but first I wanted to make sure I understand where the project sits as of right now. Could someone link here what version of the text is being used and if it is the text that is already marked up by the Pitt main students what tags will you be keeping versus getting rid of? Did you collectively decide to work within the TEI and if so do you need help finding tags for what you wish to markup? I ask all of these questions because in order to knock out some schema rules it would be best to have a list of what is being kept, what isn't, and what is being added so then we can come up with the appropriate rules and an XSLT to do an identity transformation (XSLT EXERCISE 1) that will remove the unwanted tags. If you are going with a blank text (like the Project Gutenberg text) then we need to come up with a tag set and knock the schema out first thing so a bunch of us can jump on marking it up. If someone wants to take the lead on this and just answer therse few questions listed out more clearly here I think it would be a good start to getting everyone on the same page....  
1. What text are you using?  
2. Are there tags already in the text that need removed?  
3. What do you plan on marking up to later be queried?  
4. Are you using the TEI? If so do you need help choosing logical tags for what you wish to markup? If not have you come up with element and attribute names to represent what you wish to markup?

**Comment by RJP43** [comment]
I agree with @mmm202 getting your markup and a schematron in place is most important and will help answer the rest of the midterm questions. Once you have a clear plan on what is being marked and how it is being marked you will know what it is you want to pull. The questions go hand in hand.

**Comment by RJP43** [comment]
@laurenmcguigan by extracting names and places do you mean that those are the two parts of the Pitt Main code you decided on keeping and if so then I would suggest writing an XSLT identity transformation to turn their XML into a little less marked up (and confusing) XML for your use. Then can add markup to grab the layers of the narrative in your new XML using Regex. Great job thinking of ways to use code you know to get the files prepped for additional markup. Once you guys have marked up files we can use XQuery and XSLT to extract the certain aspects of the markup that you want to create visualizations from or create HTML pages with.

**Comment by jlm323** [comment]
I think using the Oakland team's Decameron file would be good for this, since I read through the first day and I think we can work with the markup they've already done. However, I think we may need to change the elements slightly to be more accommodating. Looking at their file, do we need to keep the milestone self closing tags? And is that only for TEI? I think it would be good for us to use TEI because it seems more standard/has longevity. 

I read the First Day and took some notes on it:
- The themes of the first day seem to be about religion, virtue, and being a good person. The settings for most of the stories seem to just be in different places around Italy. One is in Egypt and one is in France for a part of it. I'm going to read Day 2 right now to see if I can come up with a good research topic.
- The main narrative is denoted by `<div3 who="author" type="commentary">` while the main characters stories are denoted by `<div3 who="filostrato" type="commentary">` for example. (Filostrato is just one of the 10 main characters.)
- As for the elements that I think need tweaking, all of the `<q direct="unspecified">`, which denotes something in quotations, doesn't make a whole lot of sense to me. I'm not sure what the attribute "unspecified" really means, so maybe changing that to "speech" or something else. The problem with this is that I've noticed some phrases or words in the `<q direct="unspecified">` tag and they don't necessarily seem to be speech.
- As for nested narratives, two stories in day one have a story contained inside but only one story is marked with `<q type="novella" direct="unspecified">` and this can be found in the 7th story on the First Day.

**Comment by RJP43** [comment]
@jlm323 

In an attempt to decode their code here are some links we might find useful in trying to figure out the markup:

Here is the [source site](http://www.brown.edu/Departments/Italian_Studies/dweb/texts/) that provided the basic markup they based their revised markup on from Brown.

This is the [final project site of the Pitt team](http://decameron.obdurodon.org/index.html), which might be worth reading through their methods. And this is their related [GitHub](https://github.com/mgb18/Boccaccio-s-Decameron). 

My only big concern with using their markup is trying to figure out how the elements line up to what you want to investigate; however, you seem to be having a good handle in understanding most of what they are using structurally to hold the different narratives. I'm not sure I entirely understand the number endings on the divs or the milestones, but if you do then great! Luckily they used TEI so we can refer to the TEI guidelines to try to get some clues as to what they meant with the `q` elements. This is what I found that may be useful regarding [quoted material](http://www.tei-c.org/release/doc/tei-p5-doc/en/html/CO.html#COHQQ).

**Comment by jlm323** [comment]
@RJP43 
The divs seem to be divided like this:

```
<div1>
    First Day
    <div2>
        Novella 1
        <div3>Author/narrator's commentary</div3>
        <div3>Character's commentary</div3>
    </div2>
    <div2>
        Novella 2
        <div3>Author/narrator's commentary</div3>
        <div3>Character's commentary</div3>
    </div2>
</div1>
```

I think if we wanted to start with totally fresh markup, it would be fine. We just need to decide on what tags to use as soon as possible. Also, I don't think the milestones are completely necessary because they just seem to be denoting a change in the text. 
Edit: I think if we chose to use the Oakland markup to look at nested narratives, we may need to create a different element or an additional element so we can actually see how it is nested. Just a thought.

**Comment by laurenmcguigan** [comment]
So are we keeping the div tags and extracting the quote tags? And what other tags can we keep? Maybe we should just make a list first of the tags we are keeping and the ones we are extracting. The we can write a schema rule that says "don't include 'old tag'" to ensure everything is extracted.

**Comment by laurenmcguigan** [comment]
Also, it doesn't look to me like names are tagged within the text.  Is that something we have to add? I see that they are all listed at the beginning of the file, but not inside the text. Is that right or am I looking at it wrong?

**Comment by jlm323** [comment]
@laurenmcguigan Yes, the names and places aren't tagged but are listed at the very beginning with some informational attributes alongside them. We may have to start tagging them. 
I think we should remove the milestone tags and keep the paragraph tags since they seem to be the original breaks in text. What tags do you think we should keep or remove?

I noticed `<pb n="22"/>` tags that continue to count up. I'm guessing those are page numbers although I'm not sure.

**Comment by laurenmcguigan** [comment]
@jlm323 I thought so! And according to XPath, there are 594 different names listed, so we might want to get started on that. But, do we want to tag ALL of the names? Type `//person` into Xpath and see if you think we should keep all of them. It doesn't seem like they are all super important, but I'm not sure because I haven't read a lot of it.  Maybe we could tag just the main characters?

**Comment by jlm323** [comment]
@laurenmcguigan I think just tagging the main characters is a good place to start! 
The main ladies are: Pampinea, Fiametta, Filomena, Emilia, Lauretta, Neifile, & Elisa.
And the main men are: Pamfilo, Filostrato, Dioneo.

**Comment by laurenmcguigan** [comment]
@jlm323 sounds good! Also, I believe pb is page break, so I also think we should keep those. I agree that we don't need the milestone tags. I'm really not quite sure what they are used for. We would also need to add the place names within the text. Is there anything else we might want to add besides people and places?

**Comment by jlm323** [comment]
@laurenmcguigan I think that's it for now. I'm going to keep reading through it and see if I can find anything else that might need tagged or changed.

**Comment by RJP43** [comment]
@jlm323 oh okay yea that structure makes sense then. So `<div1>` always holds a day, `<div2>` always holds a novella, and `<div3>` always holds the commentary of both narrator/author and other characters. Questions to ask and things to consider when preparing to create an XSLT for identity transform and schematron: 

Are the div#s ever out of that sequence of first `<div1>`, then `<div2>`, then `<div3>` and how is the prologue handled that may be different?

Are we sure the `<milestone>` doesn't mark anything of importance regarding the structure of the frame narrative (I'm not entirely sure what you (@jlm323) mean by " denoting a change in the text")?

In TEI you will want to determine whether you want to use `<placeName>` versus `<place>` as well as `<person>` versus `<persName>`. TEI documentation on all this can be found [here](http://www.tei-c.org/release/doc/tei-p5-doc/en/html/ND.html). You might also try just searching each of these element names in the [TEI Guidelines](http://www.tei-c.org/release/doc/tei-p5-doc/en/html/index.html) since that interface can be somewhat easier to navigate quickly. 

Might move the personography file outside of the main XML and into its own file with 2 simple XSLT identity transforms one that sets up the new personography file (and placeography... they can be in the same file... like the Mitford Site index and @ebeshero or I can help you with setting up a basic structure for that) and the new XML with only the certain elements you wish to keep included. 

@laurenmcguigan I think it is wise to make sure you keep paragraph `<p>` and page breaks `<pb n="#"/>` but you might want to write a rule saying when paragraphs appear (always inside a div of certain kind?) and that the page break numbers are in sequence. 

Let's figure out what the different attributes are on `<q>` and how the significance of each one might be useful to us when we go to query the information. We might also want to consider what we could add to the `<q>` elements.... is there a way right now that we can identify through xpath when a main character versus minor character is speaking, or who is having the most conversation with who else? I don't know if either of these are relevant but when you start considering things to add this might be something and then the research question could take a twist on gender or character types. 

@jlm323 You might start assigning people to specific tasks ... like @laurenmcguigan can work on the schematron maybe creating rules for some of the things we have listed above and more as they come up. @laurenmcguigan perhaps start an issue with only suggestions for schematron rules if that is what you guys decide is your task. You could assign someone else on the task of separating the two documents and removing milestone markers by doing the two XSLT transforms (whoever is doing that feel free to check out the [Nelson site index](https://github.com/RJP43/CitySlaveGirls/blob/master/webDevelopment/siteIndex.xml) to see how information for different things could be organized). And you might get someone else started on going through a specific day or number of days to mark people and places (only after you make the decision on which TEI tags to use of course.) Starting a code book sometimes is helpful to organize the whole team on decisions made about the way it has been decided to mark something (again if you need a references of what that might look like the [Nelson Codebook](https://github.com/RJP43/CitySlaveGirls/wiki/Checkout-our-Codebook!) and the [Mitford Codebook](https://docs.google.com/document/d/1r-8NGPJL1pZ20pnfvoX5OT0DkcDi-NBp5urJiZwx1sY/pub) would be two options to gain the general idea.... but basically just state the elements and what attributes correspond and what the elements and attributes are for and this could be done in a word doc or whatever ... I think a GitHub wiki easiest since you can literally rep. code because wikis are written in markdown just like issue comments). Having a codebook helps when you do a lot of the methodology write up for the project for the main site too.

**Comment by ebeshero** [comment]
@jlm323 @RJP43 The use of div1, div2, div3, etc is really not favored in TEI community (even though the Guidelines sort of support them). It's an old TEI method and the problem with it was if the divs were not actually nested correctly, the numbers would be meaningless--and worse, it's often unclear what the numbers mean to people visiting your project and viewing your code (as you found here).  So here's something you could change right away. Make those `<div>` elements be meaningful by converting them into something like:
`<div type="day">`
`<div type="story">`
etc. 

Here's some reading from the TEI Guidelines Chapter 4: Default Text Structure, with lots of examples:
http://www.tei-c.org/release/doc/tei-p5-doc/en/html/DS.html#DSDIV3

Now, here's a really serious question: I wonder if milestone elements were used in the text to signal some issue with overlapping hierarchy, or the narrative _levels_ being shifted out of order? So, for example, what if the frame narrative _interrupts_ inside an internal story?
 I don't know because I've not reviewed the Oakland team's code that closely...But `<milestone>` is certainly not the ideal element to use because there's a TEI element designed specifically _for_ that problem. It's called `<floatingText>`, and when you see an interruption in the nested layering of stories, you should consider using this: http://www.tei-c.org/release/doc/tei-p5-doc/en/html/DS.html#DSGRPF
examples: http://www.tei-c.org/release/doc/tei-p5-doc/en/html/DS.html#DSFLT
Hope that helps!

**Comment by ebeshero** [comment]
@jlm323 @laurenmcguigan @mjb232 @mmm202 @RJP43 Wow! I see that Bocaccio's _Decameron_ is a featured example on the page about the use of the `<floatingText>` element in the TEI Guidelines. So if it is NOT being used in the Oakland team's code, perhaps our team had best introduce it!
http://www.tei-c.org/release/doc/tei-p5-doc/en/html/DS.html#DSFLT

The point of `<floatingText>` is to handle shifts in the levels of a narrative like the _Decameron_ or the _Arabian Nights_. These texts present us with a "frame story" on the outside that holds the whole thing together (so the story of the the people gathered together in quarantine to sit out the plague, who entertain each other by sharing stories). The stories _within_ those stories could be said to be _nested inside_, except that sometimes, someone in the _frame_ level steps in and interrupts. (An interruption could be something shallow like, "Okay, I need a glass of water." or "There's someone knocking on the door--let me go get it..." but more likely it's built into the structure of the storytelling to make things interesting and remind us that what you're reading is delightfully complex, that there are layers in it. So the frame narration might seriously involve someone in the frame level reacting to a story being told or responding in some way to it.  

When you represent _Decameron_ online it would be useful to be able to tease out where the frame narration intersects with the internal stories. And for your collection of information about people and places, you'd want to be able to distinguish that too--what places are mentioned "out" in the frame, and what places are inside the internal stories? Just some ideas...

**Comment by jlm323** [comment]
Here's an excerpt of one of the stories including some milestones:

```
<milestone id="p01020022"/>Which matters,
 <pb n="37"/>with many others which are not to be mentioned, our modest and
 sober-minded Jew found by no means to his liking, so that, his
 curiosity being fully satisfied, he was minded to return to Paris; which
 accordingly he did. <milestone id="p01020023"/>There, on his arrival, he was met by Jehannot;
 and the two made great cheer together. Jehannot expected Abraham's
 conversion least of all things, and allowed him some days of rest before
 he asked what he thought of the Holy Father and the cardinals and
 the other courtiers. <milestone id="p01020024"/>To which the Jew forthwith replied:
```

It seems to just note a shift in the story, not necessarily an interruption in the frame narrative. 

If I'm understanding correctly, we would use floating text whenever there is a break in the frame narrative. For instance, 

```
<floatingText>
When Neifile had brought her story to a close amid the
      commendations
      of all the company, Filomena, at the queen's behest, thus
      began:
</floatingText>
```

So far, the stories are told uninterrupted besides the return to the main narrative between the stories. Also, I really prefer the `<div type="day">` and `<div type="story">` tags over the numbered div tags.

As for tasks, @laurenmcguigan has the Schematron. @mjb232 if you could move the personography and placeography into it's own file. And @mmm202 if you could start marking up the people and places for the First Day once we decide on tags for them. I created a [GitHub wiki for our codebook](https://github.com/jlm323/DecameronProject/wiki/Codebook).

**Comment by mjb232** [comment]
@jlm323 Ill get started on the Personography and Placeography right away.

**Comment by mjb232** [comment]
Also, something we might want to think about is whether or not we want to actually mark up the entire text. At this point in the game what we're doing is fine, since all of our mark-up is more or less simple and applies to every story. However, if we want to do deep analysis and comparisons, using the entire text would be impossible for the time we have, so we need to select specific stories. Now, figuring out which stories to use depends on what we are actually trying to accomplish, so we need to start considering what our purpose here is. 

One idea is that we could take a few of the main story tellers and compare their stories side-by-side. We could look for reoccurring themes, motifs, imagery, places, characters etc. We could also just do a comparison of the stories contained within a day. Each day has a certain theme and so there are bound to be some consistencies between each of the stories. 

Of course we could work with the entire text by simply comparing how many instances of specific things within each story: such as characters, places, and themes. This can be done using RegEx to mark everything up, and then XPath to get our numbers. We would then need to figure out a way to present all this information on the website.

This is just some food for thought while working, and any other ideas are welcome! We just need to find our goal here so that we can start working to achieve said goal.

**Comment by jlm323** [comment]
Good thinking! I think we should consider focusing on a few characters and compare their stories. I'd like to warn you that the character Dioneo seems to be a special case. At the end of the First Day, he asks if he can be the last story teller from then on and he asks if he can deviate from the theme of the day. So his stories do not necessarily fit in with the theme of the other stories on a given day.

**Comment by mjb232** [comment]
That's interesting actually. I'd like to take a look at Dioneo's stories and see how they match-up to one another, and to the other stories.

**Comment by mjb232** [comment]
@ebeshero  @RJP43  @jlm323 Also, I'd hate to say it but I'm not sure how to approach the Personography and Placeography. How should I format it, and should I be modifying the tags and attributes? Does anyone know of any material I could read or examples I could use to base ours after? Sorry, but I never really got a firm handle on the prosopography. Tips?

**Comment by ebeshero** [comment]
@mjb232 No worries--prosopography sounds difficult, but you just want to read something about it and see some examples to get a sense of it. 
Here's something I wrote up for the Digital Mitford project team: [Digital Mitford Explanation of Prosopography](https://docs.google.com/document/d/1r-8NGPJL1pZ20pnfvoX5OT0DkcDi-NBp5urJiZwx1sY/pub#id.7x7otqxw11d4)

And here's a nice simple working example from our Amadis project--it's kind of similar to the sort of thing you probably want to be doing with Decameron, since these are kind of similar texts: [Amadis Prosop File on GitHub](https://github.com/ebeshero/Amadis-in-Translation/blob/master/XML-and-Schematron/SI-Amadis.xml)

Actually, you may want to download that file from [here, the raw view](https://raw.githubusercontent.com/ebeshero/Amadis-in-Translation/master/XML-and-Schematron/SI-Amadis.xml) and do some XPath on it or look at it in Outline view so you have a sense of this. It's basically a file full of different kinds of lists. Every entry on each list has a distinct xml:id on it. 

@jlm323 @laurenmcguigan I'm not sure you want to actually remove every `<person>` tag just because you don't think you'll be interested in any but the main characters. But you may want to find a way to locate the characters you care about and flag them somehow--with an attribute. One simple way will be when you have a Prosopography list ready, the thing that Matt will work on. That list will contain distinct `@xml:id` identifiers for the people and places you're working with, and you'll want to point `@ref` attributes (with hashtags # + id) on elements in the Decameron text so they refer to your xml:ids in the Prosopography.

**Comment by laurenmcguigan** [comment]
@ebeshero The personography makes sense, but didn't the oakland team already make one at the beginning or is that something different? Also, while we are working on that we might want to also include in the tag whether they are a main character or a minor character.

**Comment by laurenmcguigan** [comment]
Also, I think using floating text is a good idea instead of the milestone tags, but I assume they would be placed differently than the milestone tags, we can't just replace them. So, I don't think it would be practical to add floating text to the whole document, but instead just a few stories like Matt suggested. Although do we really need to add floating text tags for the purpose of our project? I suppose if we are saying who is interupting that could be useful. But I think our focus for the project is on the characters. So I guess my question is, is floating text tags necessary? And how would they fit into the purpose of our project?

**Comment by mjb232** [comment]
See, what they have at the beginning is structured differently. Usually it looks something like this:
`<person xml:id="st">`
`<persName>`
`<surname>Triplette</surname>`
`<forename>Stacey</forename>`
`<roleName>Editor</roleName>`
`</persName>`
but what the XML has is this:
`<person id="aristippo1008" religion="pagan" sex="m" role="philosopher" origin="Atene">Aristippo`
`(X.8)</person>`
The difference lies in the fact that every description of the person is used as an attribute in the Decameron file, while the examples provided us the `xml:id` attribute and describe the person using elements. So @ebeshero, does this mean that whats provided at the beginning of the document isn't actually a Prosopography? Or is it another form of Prosopography?

**Comment by ebeshero** [comment]
@laurenmcguigan I think you all have taken a closer look at the Oakland files more recently than I have, so I'll defer to you on this. Yes, from what you're saying it looks like the `<milestone/>` elements aren't the same as `<floatingText>` at all.  

The reason you might want to "map" the structure of this narrative is because it's got stories nested inside stories. And that means that some of your characters only appear inside the frame, while others are embedded inside particular internal stories. If you have the narrative levels marked in a particular day, you could then retrieve information about which characters turn up in particular levels. 

As for the personography, definitely have a look at the Oakland team's files, because you may just want to start with theirs and modify it or build on it. I've not taken a close look at their prosopography...so I'm not sure exactly what they've done..How does it look to you guys? (I'm happy to look at it with you, too!)

**Comment by ebeshero** [comment]
@mjb232 Ohhh...I see what they did. Yes, it's an alternative way to code some prosopography information, and alas what they did doesn't look like it's really TEI conformant to me... It's not exactly wrong, but it's not quite how the TEI does things. It looks like they are confining themselves to little tiny strings of text you're allowed in an attribute value so you can never record very much.  I think you all can build something better here!

**Comment by mjb232** [comment]
Yeah, I don't like it. Alright, so I could do a very basic Prosopography, just involving the main characters. I can designate them as male/female, storyteller, and since I liked @laurenmcguigan idea I'll also designate them as the main characters. Sound good?

**Comment by ebeshero** [comment]
@mjb232 I suspect the Oakland team wasn't reading up on the use of `@xml:id` as you all have. The `@id` is an attribute we usually use in HTML, and you can use it in XML too, but it doesn't have the same punch as `@xml:id`. Let me see if I can explain the difference.

In HTML we can use an `@id` on an element, like you just did recently to make a table of contents at the top of a file and have internal links down to each little section (each Emily Dickinson poem). You're only allowed to use an `@id` once inside a file, but you could reuse it in another file in your project--it's only a matter of using it just once per unit file. You can refer to those `@id` attributes using `<a href="#id">` and build _internal links_ to let your site visitors bop down to each `@id` you've set in your text. 

In XML, you're supposed to use an `@xml:id` only once in your entire project. So if you decided to break up the Decameron text into multiple files (one per each day) and you defined a list of names with `@xml:id` attributes, you're only really supposed to represent each `@xml:id` just once anywhere, and you can write a Schematron rule to make sure that every `@ref` (or other pointer attribute with a `#`) is always pointing to an `@xml:id` that you've defined somewhere. So, across 8 XML files in a project you could have hundreds of `@ref` attributes "pointing at" just one `@xml:id` defined in your central prosopography file. When you use those `@ref` attributes, the XML isn't really _linked_ like with clickable links in HTML, but you can transform it into HTML with links, or do other things to pull those pieces together--(like make little javascript mouseovers, so when we roll over a character's name, we get a little blurb of information about them). You've got a lot of options for what you do with it in web presentation mode (for transforming into HTML), but the point is that you're writing code that pulls related information together--or _data binding_. 

The Bocaccio team put their `@id` attributes in the same file, I think--up in their header. I usually have a separate file that just contains the list of names and `@xml:id` attributes, and then write Schematron to help make sure the @ref attributes in the body of the texts we're coding always have hashtags and always point to my central prosopography file, kind of my rolladex or contact list for the project. I'm happy to bootstrap my Schematron rule for hooking a Prosopography into your main files, but for now, we just want to think about how best to handle your list of names...

**Comment by ebeshero** [comment]
Okay, I just modified that several times! Just refresh the page... ;-)

**Comment by ebeshero** [comment]
@mjb232 After the long answer...yes, I think your plan sounds good!

**Comment by mjb232** [comment]
Dope. Can you think of anything else I should use to label them?

**Comment by ebeshero** [comment]
That's kind of up to you and the team, really. One question is whether the characters go by alternative names or nicknames in the text--so often a good prosop. list will help by holding the different names and somehow setting the official one apart...

**Comment by ebeshero** [comment]
...But you're probably not ready to note alternative names yet--that usually comes with reading and marking the text. For now, what you make might be pretty simple, and you can always keep building and adding detail to it as you go.

**Comment by mjb232** [comment]
Sounds good. I just finished a very simple personography. Here's a sample:
 `<person xml:id="nea">`
                        `<persName>`
                        `<surname>Pampinea</surname>`
                           `<forename></forename>`
                          `<roleName>Story teller</roleName>`
                      `</persName>`
                     `<sex>Female</sex>`
                  `</person>`
It's in TEI (which is sooooooo easy to work with).

**Comment by ebeshero** [comment]
Looks good...but question: Is Pampinea a surname (surname = a last name)? (Do the characters have last names that you can see?) If not, surnames and forenames are optional, and you could just list her as `<persName>Pampinea</persName>`. TEI is perfectly happy with that. Also, you'll want to make sure you've opened and saved this as a `TEI ALL` file in oXygen. 

So, here's something neat about TEI personographies--neat as in tidy: The order of your names can matter, so you know where to find the most important or useful names when you hunt for them with XPath:
- The first `<persName>` should maybe be the one that matters most or is most commonly used.
- If you break down a name into multiple parts, there are only `<forename>`, `<surname>`, and `<roleName>` elements that can go inside. So, let's say someone has  first name, a middle name, and a last name, or a couple of different last names. Here's how you might write me (since my name is kind of complicated):

```
<person xml:id="ebb">
      <persName>
            <surname>Beshero-Bondar</surname>
            <forename>Elisa</forename>
            <forename>Eileen</forename>
     </persName>
     <persName type="alt'>
             <surname>Beshero</surname>
             <forename>Elisa<forename>
             <forename>Eileen<forename>
      </persName>
     <persName type="nickname">Dr. B</persName>
</person>
. . .
<note resp="#mjb232">Dr. B was this crazy professor at <placeName ref="#upg">Pitt-Greensburg</placeName>...yada yada yada...and she knew a bunch of people, including <persName ref="#mjb232>Matt Burch</persName>, <persName ref="#rjp43">Becca Parker</persName>, etc etc etc...</note>
```

TEI doesn't have a "middle name" element, so you just use a bunch of `<forename>` elements and list them in order from first name through middle names. 
You could add the other stuff in the entry too. But see how a `<note>` element could come in handy to help document relationships in your lists?

**Comment by laurenmcguigan** [comment]
So this is all I've got for the schematron so far and I don't know where to go from here
`<?xml version="1.0" encoding="UTF-8"?>
<schema xmlns:sch="http://purl.oclc.org/dsdl/schematron" queryBinding="xslt2"
    xmlns:sqf="http://www.schematron-quickfix.com/validator/process"
    xmlns="http://purl.oclc.org/dsdl/schematron">
    <ns uri="http://www.tei-c.org/ns/1.0" prefix="tei"/>
    <pattern>
        <rule context="tei:text//tei:pb[following::tei:pb]">
            <assert
                test=""> the page breaks must be in chronological order</assert>
        </rule>
        </pattern>
</schema>`

How do I say that they have to be in chronological order? I thought I could use substring, but the page breaks aren't in specific places.

**Comment by ebeshero** [comment]
@laurenmcguigan Do you mean numerical order? Or really, chronological order (by date)? _confused_

**Comment by laurenmcguigan** [comment]
Yes, I meant numerical! @ebeshero

**Comment by jlm323** [comment]
I'm going to be reading and marking up Day Two once we have the ids squared away, but I wanted to ask about the `<q>` elements: would it be beneficial to add attributes that said whether the speech is coming from a main character or a minor character? Like this:

`@who` values
who="mainChar" for showing a main character is speaking
who="minorChar" for showing a minor character is speaking

Or is that confusing because typically the main characters will be conversing at the frame narrative level in the `<floatingText>` element?
Also, I'm wondering how we can markup the stories within the stories like I mentioned before. Would using a `<div>` element with an attribute value like "nestedNarrative" work there or would it be better to include some sort of attribute value on the `<q>` elements like the Oakland team did with `<q type="novella">`?

**Comment by ebeshero** [comment]
@jlm323 Hi Jessica! We're having a confab in my office with Becca and Dr. Triplette, and here's what we thinking:
- `<q>` elements: I would not recommend adding attributes because those will require maybe a judgment call that might be tough on the project team. How can we tell a "main" character apart from a "minor" character? In a way _every_ character is kind of minor...so the distinction might just be between a character in the narrative frame vs. inside the novella (inner stories). Also, I think your point is just right: that if the `<q>` elements are inside `<floatingText>`, we'll know they're being spoken by frame characters, so I'm thinking we can hold off on tagging the `<q>` elements for now.
- As for the nesting of the inner stories, I like your idea to add a `<div type="nested">` (and I just simplified that so the attribute value isn't so long)! Now, if you notice that there are characters from an outer layer of narrative commenting _inside_ the `<div type="nested">`, you should use `<floatingText><body>....</body></floatingText>`, and think about whether the characters commenting are from the outermost "frame" narrative or from the surrounding outer novella, and we can add an `@type` attribute to `<floatingText>` to match.

**Comment by mmm202** [comment]
@jlm323 I'm just wondering if I am still marking up day one with the places and people for the project midterm?

**Comment by mjb232** [comment]
Just letting you know I'm going to start on the ids around three thirty (once I'm out of class). Anything else I need to take a look at or do?

**Comment by mjb232** [comment]
Just letting you know I'm going to start on the ids around three thirty (once I'm out of class). Anything else I need to take a look at or do?

**Comment by mjb232** [comment]
Sorry for the double post, my phone was being weird.alsooooo I'm not going to be able to attempt this until tonight. I have drive home and back. Is this going to be an issue?

**Comment by jlm323** [comment]
@mmm202 Yeah, you can continue to markup the people and places for the First Day. I will be doing it for Day Two.

@mjb232 And that's okay, Matt. As long as you can get some of it done this evening. We can always use schematron to make sure the references are pointing to the personography.

**Comment by laurenmcguigan** [comment]
@mmm202 you can mark up all the days at once using find an replace, it might be best to do them all at once.

**Comment by laurenmcguigan** [comment]
I can look up some of the minor characters if you want to do the major ones @mmm202. We just need to make sure we are doing the same thing with @mjb232 as far as ids and ref. Are the ids in the personography and then ref are in the actual tags in the body. Does that make sense?

**Comment by jlm323** [comment]
@laurenmcguigan I've just been tagging like this: `<persName>Filostrato</persName>` for the time being. And I was thinking we would add the ref to the ids later. Or should we just wait for @mjb232 to finish the whole personography before marking any names or places at all?

**Comment by mjb232** [comment]
How do I tell XSLT to keep the tags not specified in my templates?

**Comment by laurenmcguigan** [comment]
@jlm323  I mean, there really is not wrong way to do it I'm just thinking, if you go tag all the persName s now, you're just going to have to go back and find them all again to add the attributes. It's just kind of confusing because then we have to sort of wait on @mjb232 (sorry Matt!) to do the personography  before we can get started on tagging anything. Maybe we could all help Matt on the personography over the weekend?

**Comment by ebeshero** [comment]
@laurenmcguigan @mmm202 @jlm323 You can definitely tag persName and placeName and other kinds of name elements without waiting for @mjb232. You can add your `@ref` attributes to those elements later when the personography is worked out.

**Comment by ebeshero** [comment]
@mjb232 If I'm understanding your question (and I may not be), you need to write in tags that are being "consumed" in a template match. So, say you are matching on <person> elements in one XML file:

```
<xsl:template match="person">
...
</xsl:template>
```

Then, when you match on those elements, their _contents_ are what is output--the text() node and other elements inside. So if you want the `<person>` element to be there wrapping the whole, you write it into to template, like so:

```
<xsl:template match="person">
<person>
    <xsl:apply-templates/>
</person>
</xsl:template>
```

Remember how that works?

**Comment by ebeshero** [comment]
Team: It's not a problem to add tags now without @ref attributes--because those tags are an XSLT identity transform away from being updated with @ref attributes from the personography entries. The part of hunting down all the names is the sort of thing you can do with Regex, find & replace, etc. And that's a different kind of work altogether. I think you should be working on that now.

**Comment by laurenmcguigan** [comment]
@ebeshero @mmm202  @jlm323 Sorry about that!! If that is the case, would it be okay if I worked on Day 3? And @ebeshero  are those all uploaded separately or can do I need to copy and paste just day 3 into my oxygen?  What is the best way to do that so I am not already tagging what they have tagged?

**Comment by ebeshero** [comment]
I left the entire Decameron in one file because of its nested narration. I think it's okay for each of you to work on separate days, and the way to do that is to **select** the entire element holding your assigned `<div type="day">`. You can select and highlight that from the Outline view in oXygen, where you see the tree: click on your div there, and in your main oXygen window you'll see your entire div highlighted. Then if you're using Find & Replace, look in "Scope" and click the box next to "Only selected lines" and deselect "All". That way you're only doing Find & Replace over your section...

But you know what? You really shouldn't have to do that. Why not do Find & Replace over the **entire document and find ALL the Fiamettas at once**? I think each day will have a different selection of people, but the names in the frame narration _will_ overlap...so instead of simply dividing up by day, what about this?

Can one of you work on all the FRAME narration throughout the document? That would include hunting for names inside floatingText and in the prologue and opening portion, and the conclusion at the end. That person will likely tag all the frame characters. Then, the others can take on a particular novella, which will have its own characters inside. It's okay if some of you tag the same people across the entire file, because GitHub will be able to merge your work together as long as you are always pulling in changes before you push your new work. Does that make sense for a workflow?

I think if you separate out your work on framing text vs. inner stories, and everyone works on finding & replacing across the WHOLE document, you'll succeed in applying a LOT of proper name tagging very quickly!

Think of ways for Regex to help you find patterns as you search the text for proper names of various kinds. So, should one of you maybe be hunting for PLACE names, and the other for PEOPLE? A thought so you're tagging different _kinds of names_ instead of different days...

**Comment by mjb232** [comment]
Ok so this is coming along pretty smoothly, the only issue being that there are a ton of attributes in the original file that need to be converted. I need to pull up a list of the attributes used ONLY in the person list. I tried:
`distinct-values(//person/@*)`
but its giving me the individual values, not the attributes actual names. Im sure this is pretty simple, buuuuuuuuuut I cant remember how to do it. Also let it be known im working in the ListPersonPlaces document

**Comment by ebeshero** [comment]
One dangerous thing about names that @setriplette warned me about is that there will be multiple characters named Fiametta (for example) and others too--so you can't always be sure that the Fiametta you're tagging is the _same_ person until you go look at the context--is she in one of the inner stories or in the frame? What you _can_ be sure of is that Fiametta _is a persName_ and you can just wrap every instance she shows up in a `<persName>` element. You can "dedupe" (short for "deduplicate") or figure out which Fiametta is which) later on closer study of names that turn up in both the frame narration and the inner stories!

**Comment by ebeshero** [comment]
@mjb232 you need the `name()` function to return the attribute name!

**Comment by laurenmcguigan** [comment]
@ebeshero I can do the FRAME characters. Do I still label the with `<persName>` or should I use something else?

**Comment by jlm323** [comment]
You should still use `<persName>` so we can still reference it back to the personography.

**Comment by jlm323** [comment]
I'll tag the names and places for the first novella in Day Two, but I will still be reading the entire day and looking for topics that Dr. Triplette mentioned in her issue. I've been noticing that it seems like there is a lot of focus on nobles in the stories I've read so far so that might lead us somewhere!

**Comment by laurenmcguigan** [comment]
How do I do an outline view in oxygen?? @ebeshero

**Comment by mjb232** [comment]
So the original document listed all the id's simply as the name of the character, so when I do my XSLT the `xml:id` will just be their name.

**Comment by ebeshero** [comment]
@laurenmcguigan Go to Window --> Show view --> find "Outline" on the list (if it's not showing on the left side of your oXygen screen already...it usually opens by default...)

**Comment by mjb232** [comment]
Im trying to push my Personography and I am having a bad time haha. Help!

**Comment by laurenmcguigan** [comment]
@jlm323 can I add an attribute to persName on the frame characters? For example: `<persName type="frame">` So  we can find and extract ONLY the frame characters?

**Comment by laurenmcguigan** [comment]
So there are 200 floating text elements. Do I look at them each individually, or is there a way to highlight all of them?

**Comment by ebeshero** [comment]
@mjb232 Refer to https://github.com/jlm323/DecameronProject/issues/4 for help with GitHub

@laurenmcguigan @jlm323 That's a smart idea to add the attribute, BUT, remember if you want to grab every instance of someone's name in the whole file, we've been warned that sometimes characters in the inner stories share names with different characters in the frame! (Dr. Triplette suggested this might be an interesting phenomenon to trace). I wouldn't worry about adding the attribute in Replace because you just want to wrap all the persNames at this point. You can use XSLT later to add the `@type` to the persName elements only in frame sections--it'll be soooo much easier and safer to do that with a template match in XSLT than via Find & Replace!

**Comment by mjb232** [comment]
@ebeshero @RJP43 So how do I make an empty element not appear in my ids? I remember someone saying something about an IF statement, or something along those lines.

**Comment by ebeshero** [comment]
@mjb232 You'd write a **conditional statement** in XSLT to test whether the thing you're looking for is missing, and you'll likely use the`not()`function. XSL conditionals can be one simple `<xsl:if>` statement, or they can be more elaborate for lots of different conditions.  To see how to write these, scroll to the section titled "Conditionals" on this page: http://dh.obdurodon.org/xslt-basics-2.xhtml

And to output nothing, just don't apply templates--just open and close your conditional statement without anything inside.

**Comment by mjb232** [comment]
So this is what I got. I know Im grabbing the empty elements (I checked in xpath), but their not being eliminated. 

```
<xsl:template match="persName">
        <xsl:if test="/*[not(text())]"/>
    </xsl:template>
```

**Comment by RJP43** [comment]
Hey @mjb232 so I was thinking that the best way to describe this situation is that you want to output all of these attributes inside of these elements and when an attribute doesn't appear you don't want the element to appear. 

So (working from the current files that are on the main Decameron site) you have a template match on person and then you make all of these elements that call on specific parts (attributes) of the person element. 

Now to make that conditional simply surround the creation of the element with the if condition. Your test should check if the condition is met (so for example is there an age attribute on person `<xsl:if test=".[@age]">`) then output this element structure that links the attribute (`<age><xsl:apply-templates select="@age"/></age>`)

So for each attribute you would test to see it is there and then say what gets output if there:

```
<xsl:template match="person">

            <persName xml:id="{@id}" >
                <xsl:if test="./text()"><surname>
                    <xsl:apply-templates select="text()"/>
                </surname></xsl:if>

                <xsl:if test=".[@sex]"><sex>
                    <xsl:apply-templates select="@sex"/>
                </sex>

                <xsl:if test=".[@age]"><age>
                    <xsl:apply-templates select="@age"/>
                </age></xsl:if>
```

I ran this little bit of edited code, kept everything else as you had it, and the transformation removed all of the times there was empty elements for those few I changed.

**Comment by RJP43** [comment]
Of course we could write it the opposite way: if this doesn't appear then output nothing but I think it makes more sense to do it this way.

**Comment by ebeshero** [comment]
Yeah, what @RJP43 said! That makes more sense. I was thinking of not() when we were talking about it in class. But right, you really _only_ want to output something _if it is actually present_. And we are talking about attribute values, right? @mjb232

**Comment by mjb232** [comment]
I see! Yeah that makes sense! I was thinking I could do one line of code that could eliminate them all (I didnt want to have to edit all my code by hand haha)

**Comment by ebeshero** [comment]
@mjb232 Hmmm. I can imagine a way that might work, but _only_ if you already know that you want to keep the same element name as your attribute name. You'd:

Write an XSL template match on an element that has any attribute `@*` . 
For any attribute, get its name():  '@*/name()` would return "age" when @age="21" is on an element.
Use that attribute name() (whatever it is) in an XSL element constructor (you'd need to look this up), but reproduce the attribute name as an element name in your output, and convert the attribute value into the text contents of that element.

Yes, that might work. And the problem is, probably, that those attribute names aren't all legal as TEI personography elements...which is what would send you to converting _specific_ attributes into specific _new_ TEI elements.

**Comment by ebeshero** [comment]
@mjb232 It occurs to me that both me and @RJP43 may be misunderstanding what you're doing. It may not be that you're trying to match on attributes at all! I'm going back to this that you wrote:

```
<xsl:template match="persName">
        <xsl:if test="/*[not(text())]"/>
    </xsl:template>
```

This looks to me like it should work, but I wonder if you have another template rule for persName elsewhere in your XSLT? If so, it might be taking precedence over this and outputting an empty element. Would it make sense to write a predicate into your template rule on persName and says to output something, and do this?

```
<xsl:template match="persName[child::text()]">
        <whateverOutputElement>
           <xsl:apply-templates/>
        </whateverOutputElement>
    </xsl:template>
```

That doesn't use an `<xsl:if>` at all, but XSLT _should_ only output something whenever it encounters a persName that has a text() node. Would that work, or are we missing the mark?

**Comment by ebeshero** [comment]
@mjb232 and @RJP43 And of course, here is the problem with your code:

```
<xsl:template match="persName">
        <xsl:if test="/*[not(text())]"/>
    </xsl:template>
```

Remember what happens when you start a statement with a forward-slash and no dot? Your if test is reaching back up all the way up to the top of the document to see whether _any element_ has no text node.  It's more accurate just to say:

```
<xsl:template match="persName">
        <xsl:if test="self::*[not(text())]"/>
    </xsl:template>
```

or (with the dot notation):

```
<xsl:template match="persName">
        <xsl:if test=".[not(text())]"/>
    </xsl:template>
```

or more simply yet:

```
<xsl:template match="persName">
        <xsl:if test="not(text())"/>
    </xsl:template>
```

There. something here should work out!

**Comment by laurenmcguigan** [comment]
soo all the persName s are tagged within the floating text, so where should I go from there? Should I do an XSLT transformation to add attributes to the names within floating text?

**Comment by laurenmcguigan** [comment]
Although, Fiammetta is also known as Maria, and I think all those types of Names begin with i.e. ? Is that correct? and are those the names I should tag? and what should I tag them with?

**Comment by ebeshero** [comment]
@laurenmcguigan That was speedy work! :+1:  Can you paste a snippet of text here to show us what you mean about the alternative names for the same character? That is information that should be recorded in the personography file, so you can use an `@ref` attribute to identify "Maria" as "Fiametta"--But show us an example of what you're talking about. Is it frequent, and is she the only one you're seeing with something like an alternative name? Is it like a nickname for her, or what's the context?

**Comment by laurenmcguigan** [comment]
@ebeshero  It was right at the beginning in the introduction. It looks like this: 

```
<text xml:lang="EN"><front><argument><p><milestone unit="locus" corresp="p99990001"/><!--(i)-->Beginneth here the book called Decameron, otherwise
      Prince Galeotto, wherein are contained one hundred
      novels told in ten days by seven ladies and three
      young men.<!--(/i)--></p></argument><div type="prologue" corresp="proem"><head>PROEM.</head><p><milestone unit="locus" corresp="p99990002"/>'<!--(sc)-->Tis<!--(/sc)--> humane to have compassion on the afflicted; and as it
      shews well in all, so it is especially demanded of those who have had
      need of comfort and have found it in others: among whom, if any
      had ever need thereof or found it precious or delectable, I may be
      numbered; <milestone unit="locus" corresp="p99990003"/>seeing that from my early youth even to the present I
      was beyond measure aflame with a most aspiring and noble love<note>For
        <persName>Fiammetta</persName>, <!--(i)-->i.e.<!--(/i)--> Maria, natural daughter of Robert, King of
        Naples.</note>
```

It looks like it is within a `<note>` tag but the note tags contain other information as well. Also they use i.e. for a bunch of different things not just names. There are over 100 note tags.

**Comment by laurenmcguigan** [comment]
And before I forget, Agnolo Gabriello  is also known as Angel Gabriel. The code looks like this 

```
<note>Derisively for Agnolo Gabriello (the
 <!--(i)-->h<!--(/i)--> having merely the effect of
 preserving the hardness of the <!--(i)-->g<!--(/i)--> before <!--(i)-->i<!--(/i)-->), <!--(i)-->
i.e.<!--(/i)--> Angel Gabriel.</note>
```

Not sure if that is an alternate name, or just where his name might come from??

**Comment by laurenmcguigan** [comment]
Also there are 23 distinct persNames within floatingText. They are:

 Neifile

Filomena

 Dioneo

Melchisedech

Fiammetta

Emilia

Filostrato

 Lauretta

 Bergamino

Martellino

Pampinea

Guiscardo

Gerbino

 isabetta

 Chichibio

 Baronci

Isabella

 Gulfardo

 Calandrino

 Elena

 Niccolosa

 Biondello

Could some of these be the names of the storytellers. Do we already know there names?

**Comment by jlm323** [comment]
@laurenmcguigan I think this may help. These are the 10 main storytellers in the frame narrative:
Pampinea, Fiametta, Filomena, Emilia, Lauretta, Neifile, Elisa, Pamfilo, Filostrato, Dioneo.

I know a few of the names in the `<floatingText>` are in fact among the 10 main storytellers. Other names are characters within their stories.

@ebeshero I have tagged most of the `<placeName>`'s, just from looking at the list of people and places we put into the separate file. I don't think I got all of the placeNames so we might have to just tag them as we read through the stories.

**Comment by jlm323** [comment]
I also noticed that a lot of the placeNames were primarily in Italian (surprise!) and because we're working with the English version, I had to think some of the names through. For example, there was "Inghilterra" in the list of places but I know from taking Spanish that it is basically "England." So I just used Regex, looked for all instances of England, and tagged those with `<placeName>`. So everyone be aware when looking back at the list of people & places and trying to find them in the text!

**Comment by ebeshero** [comment]
Ahh, that will be a function of the Oakland team's markup. They did all of their markup on the Italian text, which makes me wonder if what we need to do is match up the English names with the Italian ones, if we're to use their prosopography lists.

**Comment by setriplette** [comment]
Just as a tip, place names are usually in word reference.com's dictionary, and if not, just google the Italian if you have a doubt. Some of them are not obvious to the eye. You might want your prosopography to list both, because some of the people familiar with this text would know the Italian set of names better than the English.

**Comment by ebeshero** [comment]
@mjb232 @laurenmcguigan @jlm323 @mmm202 @RJP43 I got Matt's XSLT working to build up the "site index" in TEI for Decameron, but it still needs a little work to pull information about places into placeography lists. It's not a LOT of work but requires some lookup of TEI tags and study of the original tagsets, which DO give us the info on English and Italian place names that @setriplette suggests above. We need to pull in a lot of that info from the Oakland list.

I also bootstrapped the team's engDecameronTEI.xml file with a project Schematron I adapted from the Mitford project so you can add `@ref` attributes to your persNames and placeNames so they point into the new site index and check to make sure they're correct. We can use a modification of this code to write XSLT or XQuery make those popup notes in HTML we were talking about on Monday...but first we need to get `@ref` attributes into your markup. Let's talk about all this today once you've pulled in my pull requests.

**Comment by ebeshero** [comment]
@ebb Some working notes from an e-mail exchange with David on a way to use XSLT to look for unmarked names to mark and automatically add `@ref` attributes. NOTE: this might be a little messy but is probably worth a try: 
- you should be able to use a few strategies to find strings in the body that are not tagged as persons, but that match person names in the personography. The simplest approach might be to use xsl:analyze-string on the text() nodes in the body. The workflow might be:
- First, scoop out all of the names you want to match in the personography and concatenate them into a regex that will look something like:

Matt|Lauren|Jess|Megan
- Then, match all text() nodes in the body that don’t have `<persName>` (or whatever else contains person names) as their parent and use xsl:analyze-string with the preceding regex. Capture the matching substring and write it into the output inside tags. You’ll want to add pointers to the personography in the elements you’re introducing; you can do a lookup as part of the transformation, or you can run the lookup separately afterwards.

**Comment by jlm323** [comment]
I found two more nested narratives in the text which I tagged by putting `<!--nested-->` at the beginning and end of the narrative. They can be found in Day 2 Novella 5 and Day 2 Novella 7. The nested narrative I showed highlighting it with JavaScript was Day 1 Novella 3.

I could not use `<div type="nested">` because the text is in `<p>` elements and it makes Oxygen angry. So I just commented it out and we can figure out a way to better tag them once we discuss it. Would `<span>` elements work here or is that better suited to our html pages?

**Comment by ebeshero** [comment]
@jlm323 How many nested narratives have we at this point? My instinct is to use `<floatingText>` for these, but we need to put an `@type` attribute for "nested". (Are there other nested narratives that we've marked with div elements? 

The whole team (or anyone working on network graphs, for example, or pulling data for graphs), will need to know any changes we're adding to the tagging of the Decameron hierarchy, since it can affect the kind of thing we can graph and map (in very good ways)! 

Also, (everyone!) don't be afraid to keep "crunching" and modifying data even after you've made a graph and mounted it on a webpage on your project website--because once you figure out how the graph should look and scale on the page, you should always be able to just add to it and change it quickly. Just make sure you're always pushing your updates through to GitHub and signalling the team loudly and clearly! **Remember to update the copy of engDecameronTEI.xml on eXist!** To do that you first have to delete the copy that's there, and then upload the new one. If you have trouble with that just let me know.

The hardest part is the first time you put it together--but revisions to your graphs and output will likely be pretty frequent as we modify the code

@laurenmcguigan @mjb232 @mmm202

**Comment by ebeshero** [comment]
@jlm323 Yes, now that I've consulted the TEI, it does seem like `<floatingText>` is really the best option here, just as it is for parts of the frame narrative that are embedded in the novella `<div>` elements. See http://www.tei-c.org/release/doc/tei-p5-doc/en/html/ref-floatingText.html

I would recommend `<floatingText type="nested">` And you could have narratives nested inside _nested_ narratives as deeply as you need to, and we could check for children / descendants of floatingText to find them! We could also number them with XSLT (going back to XSLT Exercise 1 where we did an identity transformation): You could do the same auto-numbering for your output HTML...it might be interesting to set apart nested narrations! 

One thing we need to prepare is a Methodology page for your website, where you describe the coding decisions we've made with the project in converting the document to TEI: We'll want to be clear about what elements we are using to encode the structure of the _Decameron_!

**Comment by jlm323** [comment]
We have 3 nested narratives so far. They were not tagged with `<div type="nested">` or any other tag previously. I just went in today and denoted them by putting comments around them.

**Comment by ebeshero** [comment]
@jlm323 I think it would be okay to update the code and use `<floatingText type="nested">` if you agree, and just let's make sure the team is on board with the change, and you:
1) push this to GitHub, and
2) update the copy in eXist for XQuery.
